app/Http/Controllers/AttendanceController.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Attendance;
+use Illuminate\Http\Request;
+use Illuminate\Database\QueryException;
+
+class AttendanceController extends Controller
+{
+    public function __construct()
+    {
+        $this->middleware('permission:read-attendance', ['only' => ['index']]);
+        $this->middleware('permission:create-attendance', ['only' => ['store']]);
+        $this->middleware('permission:read-attendance|read-attendance-own', ['only' => ['show']]);
+        $this->middleware('permission:update-attendance', ['only' => ['update']]);
+        $this->middleware('permission:delete-attendance', ['only' => ['destroy']]);
+    }
+    
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        $attendance = Attendance::all();
+        return response()->json(['status' => 'success', 'attendance' => $attendance]);
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
```

Comment:
The store function is going to return a validation error if an attendance swipe is recorded twice. This is not desirable as in the event the physical card swipers lose the HTTP response, they may re-upload a swipe. In this case, the "additional swipe" should be ignored, in favor of the existing one.

The API should return a 200 (OK) in this case instead of 201 (Created).

Reply:
In testing, this did not result in a validation error. I'm able to submit as many swipes as I'd like for the same event for the same GTID. I will add in de-deduplication and change the response code as requested.